### ISSUE TITLE
Enable categories other than a11y to have audit groups

### ISSUE THREAD

**Issue by rviscomi** [GoogleChrome/lighthouse#4093]
Right now only the Accessibility audit category can have groups of audits. Make this feature reusable so other categories like SEO can have groups.

Context: https://github.com/GoogleChrome/lighthouse/pull/4057#issuecomment-351805856

**Comment by paulirish** [comment]
this'll probably mean a decent refactor within report-renderer. 
done generally, it probably means that we enable the `notApplicable` group for all other categories too.

seems good!

**Comment by kdzwinel** [comment]
@paulirish I'm merging a11y category renderer with the default one. a11y renderer doesn't have the 'X Failed Audits' header. Should we:

1. have it in all audit categories,
1. remove it from all audit categories,
1. have it only for grouped audit categories (as it is now)

?

IMO it's redundant and we could skip it everywhere for consistency and to save some space. E.g. PWA section with an w/o the header:

<img width="1231" alt="screen shot 2018-01-18 at 00 59 21" src="https://user-images.githubusercontent.com/985504/35073901-78fa0da4-fbeb-11e7-8d9f-d0347e00927b.png">

**Comment by paulirish** [comment]
Yeah I like nuking the heading (and indentation). 

@vinamratasingal does that work for you?

**Comment by vinamratasingal** [comment]
I actually like keeping the failed audits section- it creates consistency between the other sections. Paul/Konrad- any reasons why you like nuking the header?

cc: @hwikyounglee in case she wants to add her perspective :)

**Comment by hwikyounglee** [comment]
Header gives a visual help to group/separate things. I suggest we keep it.

**Comment by kdzwinel** [comment]
IMO failed audits are clearly marked (by the description, ❌ and, in many cases, red debugString) and don't require any more introduction. Removing 'X Failed Audits' header saves us some horizontal space that, especially inside of DevTools, is very precious.

If we decide to keep them, we should probably still deal with the inconsistency between sections that use groups ('Accessibility', 'SEO') and sections that don't use them ('PWA', 'Best Practices'). Should we add headers to all of them?

**Comment by vinamratasingal** [comment]
I understand that the failed audits have the X, but I still think for consistency/visual help perspective, we should keep it in there. I suggest we push this paradigm across all our sections. 

cc @paulirish @hwikyounglee

**Comment by kdzwinel** [comment]
OK, I see your point👍 I made a quick change to show how the accessibility section will look like with this header:

![header](https://user-images.githubusercontent.com/985504/35075832-a4a7cc52-fbf5-11e7-9eb7-c4e96e8258de.png)

**Comment by vinamratasingal** [comment]
Can we add in a dropdown button for the Failed Audits section too? 

@hwikyounglee @paulirish does the above LGTM to you?

**Comment by kdzwinel** [comment]
By the dropdown you mean the ability to collapse the failed audits group? Yes, we can add it. ATM it's explicitly disabled.

**Comment by vinamratasingal** [comment]
Yeah... I'm also happy to put this on the next Lighthouse meeting docket if folks have strong opinions on this/would prefer to have an in-person discussion to talk through their concerns.

**Comment by patrickhulce** [comment]
Yeah discussing in next meeting sounds good, I was in favor of nuking as well :) 

Biggest reason to nuke IMO is that the extra indentation is a waste of precious space in DevTools but honestly that issue is much bigger than just this

**Comment by hwikyounglee** [comment]
If not urgent, one way is to make the formatting cleanup as a new issue assigned to me. I just created one (https://github.com/GoogleChrome/lighthouse/issues/4284), and don't know how to assign this to myself 🤔. Could anyone help? Thanks!

**Comment by kdzwinel** [comment]
I'm not sure if the header discussion should be a blocker for the main task here (I have a PR ready). WDYT about keeping the status quo for the header ('X Failed Audits' shown only for categories w/o groups) and proceeding with making groups available for the SEO category? When #4284 will be ready, I'll be happy to go back and work on the report page adjustments as a separate task.

**Comment by paulirish** [comment]
sg. let's go ahead with whats in your PR.
